sdk/eventhub/Azure.Messaging.EventHubs/MIGRATIONGUIDE.md

```diff
@@ -0,0 +1,395 @@
+
+    await partitionSender.SendAsync(eventBatch);
+}
+catch()
+{
+    // Handle the exception here
+}
+finally
+{
+    await eventHubClient.CloseAsync();
+    await partitionSender.CloseAsync();
+}
+```
+
+In v5:
+```csharp
+var connectionString = "<< CONNECTION STRING FOR THE EVENT HUBS NAMESPACE >>";
+var eventHubName = "<< NAME OF THE EVENT HUB >>";
+
+await using (var producerClient = new EventHubProducerClient(connectionString, eventHubName))
+{
+    var batchOptions = new CreateBatchOptions() { PartitionId = "my-partition-id" };
+    using EventDataBatch eventBatch = await producerClient.CreateBatchAsync(batchOptions);
+    eventBatch.TryAdd(new EventData(Encoding.UTF8.GetBytes("First")));
+    eventBatch.TryAdd(new EventData(Encoding.UTF8.GetBytes("Second")));
+    
+    await producerClient.SendAsync(eventBatch);
+}
+```
+
+### Migrating code from `EventHubClient` to `EventHubProducerClient` for publishing events using automatic routing
+
+In v4, events could be published to an Event Hub that allowed the service to automatically route events to an available partition.
+
+In v5, automatic routing occurs when an [`EventDataBatch`](https://docs.microsoft.com/en-us/dotnet/api/azure.messaging.eventhubs.eventdatabatch?view=azure-dotnet-preview) is created using [`CreateBatchAsync()`](https://docs.microsoft.com/en-us/dotnet/api/azure.messaging.eventhubs.eventhubproducerclient.createbatchasync?view=azure-dotnet-preview#Azure_Messaging_EventHubs_EventHubProducerClient_CreateBatchAsync_System_Threading_CancellationToken_).
+
+In v4:
+```csharp
+var connectionString = "<< CONNECTION STRING FOR THE EVENT HUBS NAMESPACE >>";
+var eventHubName = "<< NAME OF THE EVENT HUB >>";
+
+var connectionStringBuilder = new EventHubsConnectionStringBuilder(connectionString){ EntityPath = eventHubName }; 
+var eventHubClient = EventHubClient.CreateFromConnectionString(connectionStringBuilder.ToString());
+try
+{
+    EventDataBatch eventBatch = partitionSender.CreateBatch();
+    eventBatch.TryAdd(new EventData(Encoding.UTF8.GetBytes("First")));
+    eventBatch.TryAdd(new EventData(Encoding.UTF8.GetBytes("Second")));
+
+   await partitionSender.SendAsync(eventBatch);
+}
+catch()
+{
+    // Handle the exception here
+}
+finally
+{
+    await eventHubClient.CloseAsync();
+}
+```
+
+In v5:
+```csharp
+var connectionString = "<< CONNECTION STRING FOR THE EVENT HUBS NAMESPACE >>";
+var eventHubName = "<< NAME OF THE EVENT HUB >>";
+
+await using (var producerClient = new EventHubProducerClient(connectionString, eventHubName))
+{
+    using EventDataBatch eventBatch = await producerClient.CreateBatchAsync();
+    eventBatch.TryAdd(new EventData(Encoding.UTF8.GetBytes("First")));
+    eventBatch.TryAdd(new EventData(Encoding.UTF8.GetBytes("Second")));
+    
+    await producerClient.SendAsync(eventBatch);
+}
+```
+
+### Migrating code from `EventHubClient` to `EventHubProducerClient` for publishing events with partition key
```

Comment:
Can we add migration from SendAsync(events, partitionKey) API as well?

Reply:
Sounds good!

I've already used batch send for `Migrating code from PartitionSender to EventHubProducerClient for publishing events to a partition` example, I'll update this one to use `SendAsync(events, partitionKey)`.